engine.go

```diff
@@ -0,0 +1,247 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"os/user"
+	"runtime"
+	"strings"
+	"sync"
+	"syscall"
+
+	"github.com/coduno/app/models"
+	"github.com/coduno/app/util"
+	"github.com/coduno/piper"
+	"github.com/m4rw3r/uuid"
+)
+
+var (
+	// keep both home dir and docker home dir even if they are the same because
+	// on windows we need to have a dockerHomeDir=/c/Users/$USER/tmp and a
+	// homeDir=C:/Users/%USER%/tmp. On Linux they are homeDir=dockerHomeDir=$HOME/tmp
+	homeDir       string
```

Comment:
I do not see why we store the user's home directory here. We should rather focus on providing two different kinds of *temporary directories*. Once for configurations, and one acting as Docker volume.